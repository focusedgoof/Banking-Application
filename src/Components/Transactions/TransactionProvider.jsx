import React,{ createContext, useState} from 'react';
import PropTypes from 'prop-types';

export const TransactionContext = createContext();
 
export const TransactionProvider = ({ children }) => {
  const [transactions, setTransactions] = useState([]);

  const addTransaction = (newTransaction) => {
    setTransactions([...transactions, newTransaction]);
  };
 
  return (
    <TransactionContext.Provider value={{ transactions, addTransaction }}>
      {children}
    </TransactionContext.Provider>
  );
};

TransactionProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TransactionProvider;
