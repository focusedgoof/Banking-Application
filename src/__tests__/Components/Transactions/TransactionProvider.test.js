import React from 'react';
import { render } from '@testing-library/react';
import TransactionProvider from '../../../Components/Transactions/TransactionProvider';

test('renders TransactionProvider component snapshot', () => {
  const { asFragment } = render(<TransactionProvider />);
  expect(asFragment()).toMatchSnapshot();
});
