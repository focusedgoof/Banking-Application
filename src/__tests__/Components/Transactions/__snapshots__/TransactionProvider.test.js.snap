// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TransactionProvider component snapshot 1`] = `<DocumentFragment />`;
